kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/exception/BigQueryStorageWriteApiErrorResponses.java

```diff
@@ -75,4 +99,29 @@ public static boolean hasInvalidSchema(Collection<String> messages) {
     public static boolean isStreamClosed(String errorMessage) {
         return errorMessage.contains(STREAM_CLOSED);
     }
+
+    /**
+     * Tells if the exception is of storage exception type and  error code belong to the list of non retriable
+     * storage error code.
+     *
+     * @param exception Exception received from Batch mode data ingestion
+     * @return Retruns true if the exception is non-retriable
+     */
+    public static boolean isNonRetriableStorageError(Exception exception) {
+        Exceptions.StorageException storageException = null;
+        Throwable t = exception.getCause();
+        if (t instanceof StatusRuntimeException || t instanceof StatusException) {
+            storageException = Exceptions.toStorageException(exception);
+        }
+        if (storageException == null) {
+            // it is not a storage exception. We will consider it something unknown and thus non-retriable
+            return true;
+        }
+        String errorCode = storageException.getStatus().getCode().name();
+
+        logger.trace("Storage exception occurred with errorCode {} and errors {} ", errorCode, storageException.getErrors().toString());
+
+        return Arrays.asList(nonRetriableStreamFailureCodes).contains(errorCode);
```

Comment:
nonRetriableStreamFailureCodes can be represented as a Set if we want only a membership test